From 903645a0a627c86d68b6685ddb78855d54e18b3f Mon Sep 17 00:00:00 2001
From: Stefan Seyfried <seife@tuxbox-git.slipkontur.de>
Date: Sun, 4 Nov 2012 18:08:26 +0100
Subject: [PATCH 6/7] aotom: add additional chars for VFD, fix missing chars

* add "{|}" chars, fix "_" char, fix display of "+_" chars.
* improve "K" and "0" readability
* add comments to the character map to ease understanding the data.
---
 frontcontroller/aotom/aotom_vfd.c |  137 ++++++++++++++++++++-----------------
 1 file changed, 75 insertions(+), 62 deletions(-)

diff --git a/frontcontroller/aotom/aotom_vfd.c b/frontcontroller/aotom/aotom_vfd.c
index e92b9db..e159903 100644
--- a/frontcontroller/aotom/aotom_vfd.c
+++ b/frontcontroller/aotom/aotom_vfd.c
@@ -219,70 +219,77 @@ u8 YWPANEL_CharArray[]=
 	0x7B,0x11,0x76,0x75,0x1D,0x6D,0x6F,0x31,0xFF,0x7D/* 0~9*/
 };
 
-//     a
-//f h j k b
-//  g i m
-//e r p n c
-//     d
+//  aaaaa
+// fh j kb
+// f hjk b
+//  ggimm
+// e rpn c
+// er p nc
+//  ddddd
 //a b c d e f g h
 
+/*               7 6 5 4 3 2 1 0 */
 /*address 0 8bit g i m c r p n e */
-/*address 1 7bit d a b f k j h    */
-
-u8 CharLib[48][2] =
-{
-	{0xF1, 0x38},	//A
-	{0x74, 0x72},	//B
-	{0x01, 0x68},	//C
-	{0x54, 0x72},	//D
-	{0xE1, 0x68},	//E
-	{0xE1, 0x28},	//F
-	{0x71, 0x68},	//G
-	{0xF1, 0x18},	//H
-	{0x44, 0x62},	//I
-	{0x45, 0x22},	//J
-	{0x46, 0x06},	//K
-	{0x01, 0x48},	//L
-	{0x51, 0x1D},	//M
-	{0x53, 0x19},	//N
-	{0x11, 0x78},	//O
-	{0xE1, 0x38},	//P
-	{0x13, 0x78},	//Q
-	{0xE3, 0x38},	//R
-	{0xF0, 0x68},	//S
-	{0x44, 0x22},	//T
-	{0x11, 0x58},	//U
-	{0x49, 0x0C},	//V
-	{0x5B, 0x18},	//W
-	{0x4A, 0x05},	//X
-	{0x44, 0x05},	//Y
-	{0x48, 0x64},	//Z
+/*address 1 7bit   d a b f k j h */
+
+
+u8 CharLib[0x33][2] =
+{
+	{0xF1, 0x38},	//A	00
+	{0x74, 0x72},	//B	01
+	{0x01, 0x68},	//C	02
+	{0x54, 0x72},	//D	03
+	{0xE1, 0x68},	//E	04
+	{0xE1, 0x28},	//F	05
+	{0x71, 0x68},	//G	06
+	{0xF1, 0x18},	//H	07
+	{0x44, 0x62},	//I	08
+	{0x45, 0x22},	//J	09
+	{0xC3, 0x0C},	//K	0A
+	{0x01, 0x48},	//L	0B
+	{0x51, 0x1D},	//M	0C
+	{0x53, 0x19},	//N	0D
+	{0x11, 0x78},	//O	0E
+	{0xE1, 0x38},	//P	0F
+	{0x13, 0x78},	//Q	10
+	{0xE3, 0x38},	//R	11
+	{0xF0, 0x68},	//S	12
+	{0x44, 0x22},	//T	13
+	{0x11, 0x58},	//U	14
+	{0x49, 0x0C},	//V	15
+	{0x5B, 0x18},	//W	16
+	{0x4A, 0x05},	//X	17
+	{0x44, 0x05},	//Y	18
+	{0x48, 0x64},	//Z	19
 	/* A--Z  */
 
-	{0x01, 0x68},
-	{0x42, 0x01},
-	{0x10, 0x70},	//
-	{0x43, 0x09},	//
-	{0xE0, 0x00},	//
-	{0xEE, 0x07},	//
-	{0xE4, 0x02},	//
-	{0x50, 0x00},	//
-	{0xE0, 0x00},	//
-	{0x05, 0x00},	//
-	{0x48, 0x04},	//
-
-	{0x11, 0x78},	//
-	{0x44, 0x02},	//
-	{0xE1, 0x70},	//
-	{0xF0, 0x70},	//
-	{0xF0, 0x18},	//
-	{0xF0, 0x68},	//
-	{0xF1, 0x68},	//
-	{0x10, 0x30},	//
-	{0xF1, 0x78},	//
-	{0xF0, 0x78},	//
+	{0x01, 0x68},	// [	1A
+	{0x42, 0x01},	// \	1B
+	{0x10, 0x70},	// ]	1C
+	{0x43, 0x09},	// |\	1D
+	{0x00, 0x40},	// _	1E
+	{0xEE, 0x07},	// *	1F
+	{0xE4, 0x02},	// +	20
+	{0x50, 0x00},	// .,	21
+	{0xE0, 0x00},	// -	22
+	{0x05, 0x00},	// ,,	23
+	{0x48, 0x04},	// /	24
+
+	{0x51, 0x78},	// 0	25
+	{0x44, 0x02},	// 1	26
+	{0xE1, 0x70},	// 2	27
+	{0xF0, 0x70},	// 3	28
+	{0xF0, 0x18},	// 4	29
+	{0xF0, 0x68},	// 5	2A
+	{0xF1, 0x68},	// 6	2B
+	{0x10, 0x30},	// 7	2C
+	{0xF1, 0x78},	// 8	2D
+	{0xF0, 0x78},	// 9	2E
 	/* 0--9  */
-	{0x00, 0x00}
+	{0x00, 0x00},	// ' '	2F
+	{0x44, 0x02},	// |	30
+	{0x42, 0x04},	// <{	31
+	{0x48, 0x01},	// >}	32
 };
 
 u8 NumLib[10][2] =
@@ -1965,12 +1972,18 @@ static u8 ywpanel_vfd_map[0x80] =
 {
 	0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f,
 	0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f,
-	0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x1f, 0x2f, 0x21, 0x22, 0x23, 0x24,
-	0x25, 0x26, 0x27, 0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f,
+	//     !     "      #     $     %     &     '     (     )     *     +     ,     -     .     /
+	0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f, 0x1a, 0x1c, 0x1f, 0x20, 0x21, 0x22, 0x23, 0x24,
+	// 0    1     2     3     4     5     6     7     8     9     :     ;     <     =     >     ?
+	0x25, 0x26, 0x27, 0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f, 0x2f, 0x31, 0x2f, 0x32, 0x2f,
+	// @    A     B     C     D     E     F     G     H     I     J     K     L     M     N     O
 	0x2f, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e,
-	0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1f,
+	// P    Q     R     S     T     U     V     W     X     Y     Z     [     \     ]     ^     _
+	0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e,
+	// `    a     b     c
 	0x2f, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e,
-	0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x2f, 0x2f, 0x2f, 0x2f, 0x2f
+	// p                                                          z     {     |     }     ~
+	0x0f, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x31, 0x30, 0x32, 0x2f, 0x2f
 };
 
 void YWPANEL_VFD_DrawChar(char c, u8 position)
-- 
1.7.10.4

